web-registry/src/pages/ProjectEdit/hooks/useProjectEditSubmit.tsx

```diff
@@ -24,7 +25,13 @@ type Props = {
   }: OnTxSuccessfulProps) => void;
 };
 
-export type UseProjectEditSubmitParams = (metadata: any) => Promise<void>;
+export type NestedPartial<K> = {
+  [attr in keyof K]?: K[attr] extends object ? NestedPartial<K[attr]> : K[attr];
+};
```

Comment:
Nice one!
Could this type be reusable elsewhere?

Reply:
maybe it could be useful in credit class / credit batch creation forms I guess once we decide to work back on these ones...
do you think we should put this in `types` instead?

Reply:
In that case yes!

Reply:
done